File: 프로그래머스/0/181913. 문자열 여러 번 뒤집기/문자열 여러 번 뒤집기.java
class Solution {
    public String solution(String my_string, int[][] queries) {
        String answer = "";
        for(int i = 0; i < queries.length; i++){
            String pre = my_string.substring(0, queries[i][0]);
            String suf = my_string.substring(queries[i][1]+1);
            StringBuilder temp = new StringBuilder(my_string.substring(queries[i][0], queries[i][1] + 1));
            temp.reverse();
            my_string = pre + temp.toString() + suf;
        }
        return my_string;
    }
}

File: 프로그래머스/0/181913. 문자열 여러 번 뒤집기/문자열 여러 번 뒤집기.py
def solution(my_string, queries):
    ss = list(my_string)
    for q in queries:
        s, e = q
        while s < e:
            temp = ss[s]
            ss[s] = ss[e]
            ss[e] = temp
            s += 1
            e -= 1
    return ''.join(ss)
        
        